wiring/src/spark_wiring_ble.cpp

```diff
@@ -702,10 +702,14 @@ size_t BleAdvertisingData::serviceUUID(BleAdvertisingDataType type, BleUuid* uui
     for (size_t i = 0; i < selfLen_; i += (offset + adsLen)) {
         adsLen = locate(&selfData_[i], selfLen_ - i, type, &offset);
         if (adsLen > 0 && found < count) {
-            if (adsLen == 4) { // length field + type field + 16-bits UUID
-                uuids[found++] = (uint16_t)selfData_[i + offset + 2] | ((uint16_t)selfData_[i + offset + 3] << 8);
-            } else if (adsLen == 18) {
-                uuids[found++] = &selfData_[i + offset + 2];
+            if (type == BleAdvertisingDataType::SERVICE_UUID_16BIT_MORE_AVAILABLE || type == BleAdvertisingDataType::SERVICE_UUID_16BIT_COMPLETE) {
+                for(size_t array = 0; array < (adsLen - 2) / BLE_SIG_UUID_16BIT_LEN; array++) {
+                    uuids[found++] = (uint16_t)selfData_[i + offset + array * BLE_SIG_UUID_16BIT_LEN + 2] | ((uint16_t)selfData_[i + offset + array * BLE_SIG_UUID_16BIT_LEN + 3] << 8);
```

Comment:
One more thing: we may step over the `uuids` bounds here by going over `count` with `found`. So an additional condition will need to be added to get out of loop if `found >= count`. Same with 128-bit case.